test/Microsoft.AspNetCore.Mvc.FunctionalTests/FiltersTest.cs

```diff
@@ -84,7 +84,7 @@ public FiltersTest(MvcTestFixture<FiltersWebSite.Startup> fixture)
             var response = await Client.GetAsync("http://localhost/Anonymous/GetHelloWorld");
 
             // Assert
-            Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
```

Comment:
I think so, your BlockAnonymous filter is kicking in which means you want a 403 Forbidden rather than 401 correct?